pkg/devfile/parser/data/common/types.go

```diff
@@ -1,210 +1,449 @@
+	CustomProjectSourceType DevfileProjectSourceType = "Custom"
 )
 
-var SupportedDevfileProjectTypes = []DevfileProjectType{DevfileProjectTypeGit}
-
-// -------------- Supported devfile component types ------------ //
-// DevfileComponentType stores valid devfile component types
 type DevfileComponentType string
 
 const (
-	DevfileComponentTypeCheEditor   DevfileComponentType = "cheEditor"
-	DevfileComponentTypeChePlugin   DevfileComponentType = "chePlugin"
-	DevfileComponentTypeDockerimage DevfileComponentType = "dockerimage"
-	DevfileComponentTypeKubernetes  DevfileComponentType = "kubernetes"
-	DevfileComponentTypeOpenshift   DevfileComponentType = "openshift"
+	ContainerComponentType  DevfileComponentType = "Container"
+	KubernetesComponentType DevfileComponentType = "Kubernetes"
+	OpenshiftComponentType  DevfileComponentType = "Openshift"
+	PluginComponentType     DevfileComponentType = "Plugin"
+	VolumeComponentType     DevfileComponentType = "Volume"
+	CustomComponentType     DevfileComponentType = "Custom"
 )
 
-// -------------- Supported devfile command types ------------ //
 type DevfileCommandType string
 
 const (
-	DevfileCommandTypeInit  DevfileCommandType = "init"
-	DevfileCommandTypeBuild DevfileCommandType = "build"
-	DevfileCommandTypeRun   DevfileCommandType = "run"
-	DevfileCommandTypeDebug DevfileCommandType = "debug"
-	DevfileCommandTypeExec  DevfileCommandType = "exec"
+	ExecCommandType         DevfileCommandType = "Exec"
+	VscodeTaskCommandType   DevfileCommandType = "VscodeTask"
+	VscodeLaunchCommandType DevfileCommandType = "VscodeLaunch"
+	CompositeCommandType    DevfileCommandType = "Composite"
+	CustomCommandType       DevfileCommandType = "Custom"
 )
 
-// ----------- Devfile Schema ---------- //
-type Attributes map[string]string
+// CommandGroupType describes the kind of command group.
+// +kubebuilder:validation:Enum=build;run;test;debug
+type DevfileCommandGroupType string
 
-type ApiVersion string
+const (
+	BuildCommandGroupType DevfileCommandGroupType = "build"
+	RunCommandGroupType   DevfileCommandGroupType = "run"
+	TestCommandGroupType  DevfileCommandGroupType = "test"
+	DebugCommandGroupType DevfileCommandGroupType = "debug"
+	// To Support V1
+	InitCommandGroupType DevfileCommandGroupType = "init"
+)
 
+// Metadata Optional metadata
 type DevfileMetadata struct {
 
-	// Workspaces created from devfile, will use it as base and append random suffix.
-	// It's used when name is not defined.
-	GenerateName *string `yaml:"generateName,omitempty" json:"generateName,omitempty"`
+	// Optional devfile name
+	Name string `json:"name,omitempty"`
 
-	// The name of the devfile. Workspaces created from devfile, will inherit this
-	// name
-	Name *string `yaml:"name,omitempty" json:"name,omitempty"`
+	// Optional semver-compatible version
+	Version string `json:"version,omitempty"`
 }
 
-// Description of the projects, containing names and sources locations
+// CommandsItems
```

Comment:
Could you please update the descriptions for all the exported items in this file?

Reply:
Done.